nifi-nar-bundles/nifi-salesforce-bundle/nifi-salesforce-processors/src/main/java/org/apache/nifi/processors/salesforce/QuerySalesforceObject.java

```diff
@@ -560,6 +566,27 @@ private SalesforceSchemaHolder getConvertedSalesforceSchema(String sObject, Stri
         }
     }
 
+    private void handleError(ProcessSession session, FlowFile originalFlowFile, AtomicBoolean isOriginalTransferred, List<FlowFile> outgoingFlowFiles,
+                             Exception e, String errorMessage) {
+        if (originalFlowFile != null) {
+            session.transfer(originalFlowFile, REL_FAILURE);
+            isOriginalTransferred.set(true);
+        }
+        getLogger().error(errorMessage, e);
+        session.remove(outgoingFlowFiles);
+        outgoingFlowFiles.clear();
+    }
+
+    private StateMap getState(ProcessContext context) {
+        StateMap state;
+        try {
+            state = context.getStateManager().getState(Scope.CLUSTER);
```

Comment:
I think we should use `session.setState` instead.
That way if the session is rolled back the state change is rolled back as well.

Reply:
I agree but there's a different JIRA for that.

Reply:
Which JIRA would that be?